wsd/DocumentBroker.cpp

```diff
@@ -1491,9 +1491,33 @@ void DocumentBroker::handleUploadToStorageResponse(const StorageBase::UploadResu
             Poco::URI::encode(filename, "", encodedName);
             const std::string filenameAnonym = LOOLWSD::anonymizeUrl(filename);
 
+            constexpr const char* needle = "?access_token";
+            std::size_t index = getPublicUri().toString().find(needle);
+            std::string url1, url2;
+            if (index != std::string::npos)
+            {
+
+                url1 = getPublicUri().toString().substr(0, index);
+                url2 = url.substr(0, index);
+            }
+            else
+            {
+                url1 = url;
+                url2 = url;
+            }
+
             std::ostringstream oss;
             oss << "renamefile: " << "filename=" << encodedName << " url=" << url;
             broadcastMessage(oss.str());
+            if (!strcasecmp(url1.c_str(), url2.c_str()))
+            {
+                // WOPISrc has not changed
+                // However, due to rename, timestamp in the storage has changed
+                // and any further upload requests will result to a conflict
+                // Do notify the client about the conflict
+                // This will make the client reload the document properly
+                broadcastMessage("close: documentconflict");
```

Comment:
Broadcasting `renamefile:` command followed with `close:` is not ideal. The client might actually not process the `close:` for some time after getting `renamefile:` and cause other issues/bugs. Why can't we send only one of these? (Either `renamefile:` or `close:`, but not both.)